uniffi_bindgen/src/interface/enum_.rs

```diff
@@ -202,6 +202,28 @@ impl Enum {
         &self.variants
     }
 
+    // Get the literal value to use for the specified variant's discriminant.
+    // Tries to follow Rust's rules when mixing specified and unspecified values,
+    // but only promises to get specified values correct.
```

Comment:
It seems like it follows the logic to me, which cases would it get wrong?  If there are cases where we get it wrong, we should add a note in the docs about this.

Also, even if this doesn't match Rust's rules, each discriminant must be unique.   Otherwise the foreign enums will probably not work.  Can you add a note about that and maybe even check in the code?

Reply:
> It seems like it follows the logic to me, which cases would it get wrong? 

I read the Rust docs but not the implementation. The docs aren't as precise as I'd like, hence me trying to not oversell this implementation. I guess I should just have a bit more faith :)

> Also, even if this doesn't match Rust's rules, each discriminant must be unique. Otherwise the foreign enums will probably not work.

I should add a comment about this, but my assumption here is that we can leave that checking to Rust. If there is a variant with a duplicate discriminate then we can just assume the Rust compilation will fail, therefore we can assume there are no dupes?

> Can you add a note about that and maybe even check in the code?

I wanted to avoid that is it would mean I need to check *every* variant even when just checking the first instance, and probably would need a hashmap to remember what I'd already seen. It seems far far simpler to assume the Rust compiler checks this.

WDYT?

Reply:
Oh, the other thing it doesn't check is "overflow" - eg, the Rust compiler will complain if a `repr(u8)` overflows (eg, when one variant specified `0xFF` and the next variant specifies no value). That code doesn't even *know* the repr(), so that too is something I'm fine assuming the Rust compiler enforces.